It seems the timeout for the distributed tests is set to low and spurious failures can be seen
Increase it by a factor of 30 similar to torch/testing/_internal/distributed/distributed_test.py

Author: Alexander Grund (TU Dresden)

diff --git a/torch/testing/_internal/common_distributed.py b/torch/testing/_internal/common_distributed.py
index 4f36b31a23..fbd2b9364a 100644
--- a/torch/testing/_internal/common_distributed.py
+++ b/torch/testing/_internal/common_distributed.py
@@ -160,7 +160,7 @@ def skip_if_win32():
         "This unit test case is not supportted on Windows platform",
     )
 
-TIMEOUT_DEFAULT = 100
+TIMEOUT_DEFAULT = 3000
 TIMEOUT_OVERRIDE = {"test_ddp_uneven_inputs": 400}
 
 
